Gems/Profiler/Code/Source/ImGuiCpuProfiler.cpp

```diff
@@ -116,12 +120,18 @@ namespace Profiler
         }
     } // namespace CpuProfilerImGuiHelper
 
+    ImGuiCpuProfiler::ImGuiCpuProfiler()
+    {
+        // thread IDs are hashed internally to unify display across platforms
+        m_mainThreadId = AZStd::hash<AZStd::thread_id>{}(AZStd::this_thread::get_id());
```

Comment:
Eventually it may be useful for us to have a reverse lookup of thread name to id

Reply:
Funny you mention it, I had started looking into that yesterday after posting this.